Refactor recipe

niranjchandrasekaran opened the pull request:
- The parameters of the profiling pipeline will now reside in the data repo ([example config file](https://github.com/jump-cellpainting/jamboree-data/blob/main/config.yml)). 
- `environment.yml` has also been moved to the data repo.
- The new recipe output files identical to the old recipe ([compared here](https://github.com/jump-cellpainting/jamboree-data))

Discussion - https://github.com/jump-cellpainting/pilot-data/issues/13

niranjchandrasekaran committed:
refactor recipe

niranjchandrasekaran committed:
fix bug where all batches were not being read

niranjchandrasekaran committed:
add external annotations

niranjchandrasekaran committed:
add cell id and control processing at the plate level

niranjchandrasekaran committed:
set cmap format requiremnt to false

niranjchandrasekaran committed:
fixed metadata annotation error

niranjchandrasekaran committed:
debug

niranjchandrasekaran committed:
debug

niranjchandrasekaran committed:
change dmso to "negcon"

niranjchandrasekaran committed:
add config file from command line

gwaygenomics commented:
once the `refactor_recipe` branch is merged into `master` of the `jump-cellpainting` fork, is the forked code ready to be merged back into cytomining upstream?

I'll make documentation enhancements to cytomining upstream

gwaygenomics commented:
one very quick thing I noticed @niranjchandrasekaran - can you run [`black`](https://github.com/psf/black) on all the updated scripts?

niranjchandrasekaran committed:
run black

shntnu committed:
Create environment.yml

Adding this but will add instructions that the environment name should be changed before creating the environment

shntnu commented:
(DRAFT; will continue editing this)

> * The parameters of the profiling pipeline will now reside in the data repo ([example config file](https://github.com/jump-cellpainting/jamboree-data/blob/main/config.yml)).

Can we have an example config file in this repo?

> * `environment.yml` has also been moved to the data repo.

Any concerns with retaining this file in this repo? The only issue I see is that the environment name may need to be updated each time. But I think that's fine. We can add instructions to clarify that.

shntnu commented:
> Can we have an example config file in this repo?
> 
> >

@gwaygenomics recommends that we have this in https://github.com/cytomining/profiling-recipe instead, and not in the recipe. The recipe README can point to this config file. But the recipe repo should not contain the config.

gwaygenomics commented:
☝️ @shntnu - typo above: we should have the example in the [data repo template](https://github.com/cytomining/profiling-template) not the recipe. The recipe README will point to the example config in the template.

gwaygenomics commented:
One takeaway from yesterday for me is that there are a lot of repos to keep track of.

I propose that we keep track of the recipe and template repo from the `cytomining` org. JUMP is the only project using them in production now, so we'll develop them primarily for this project. Things can get needlessly complicated at this stage if we diverge jump-cellpainting from cytomining too much.

Concretely, this means merging this PR (LGTM) into jump-cellpainting master, then create a new PR from jump-cellpainting master to cytomining upstream/master.

I will document all of the appropriate action items in cytomining/profiling-recipe.

gwaybio commented on profiles/profile.py:
```diff
@@ -1,155 +1,156 @@
+    if annotate_steps["perform"]:
+        if annotate_steps["external"]:
+            external_df = pd.read_csv(
+                pathlib.PurePath(".", "metadata", "moa", annotate_steps["external"]),
```
the `annotate_steps["external"]` must include a relative path without having to rely on `pathlib.PurePath(".", "metadata", "moa")`

niranjchandrasekaran committed:
run black; set Assay barcode name dtype to str

niranjchandrasekaran committed:
run black; set Assay barcode name dtype to str

niranjchandrasekaran committed:
cosmetic changes

niranjchandrasekaran commented:
https://github.com/jump-cellpainting/profiling-recipe/blob/551b85f4fa9bf8efedc118c745bc8d46e5fc94a5/profiles/profile.py#L94-L96

The above lines fails when the name of the plate is an `int` instead of `str`. Hence I have explicitly set `Assay_Plate_Barcode` to `str` in the following lines

https://github.com/jump-cellpainting/profiling-recipe/blob/551b85f4fa9bf8efedc118c745bc8d46e5fc94a5/profiles/profile.py#L45-L46

niranjchandrasekaran committed:
remove duplicate

niranjchandrasekaran committed:
move step specific code within the if statements

niranjchandrasekaran commented:
While trying to run the pipeline without the `annotate` step (by setting `perform` to `false` in the config file) I found that a lot of code that was specific to annotate was outside the `if` statement that checks whether to perform that action or not. I have now moved (https://github.com/jump-cellpainting/profiling-recipe/pull/14/commits/96747a59f179144a8717afeee116b1b07375e630) action specific code in `profile.py` inside the respective `if` code blocks.

gwaybio commented on profiles/profile.py:
```diff
@@ -1,155 +1,161 @@
+            float_format=float_format,
+            compression=compression,
+        )
+        if normalize_steps["negcon"]:
```
I don't think that we'll have a good solution to support a block execution design before we merge this PR.

As an intermediate and sustainable step, I propose the following:

1. Drop the `normalize_steps["negcon"]` (and the `feature_select_steps["negcon"]`) option
2. Create two config yaml files (one with normalize -> samples set to `"Metadata_control_type == 'negcon'"`, and the other with normalize -> samples set to "all"). We can avoid rerunning aggregation by setting perform = False in the second executed config file
3. Extract out [line 23](https://github.com/jump-cellpainting/profiling-recipe/pull/14/files#diff-b9d876056b6394f85313599f3b2cf8500f517537c9a8f519afb74c6b13712f0aR23) that set filename suffixes into the configuration file
4. Add functionality to `profiles/profile_utils.py` in order to parse the new file names in the config file

gwaygenomics commented:
@niranjchandrasekaran @shntnu - what do you think about my proposal? https://github.com/jump-cellpainting/profiling-recipe/pull/14#discussion_r551597902

I'd like to get this merged in so we can continue to build upon it. If we're worried about the next data push, maybe instead of merging to `master`, we can merge to a new `develop` branch?

niranjchandrasekaran commented:
👍  I think that works as a good intermediate step. Also moving the file names out of `profile.py` will be important for making the pipeline modular in the future.

shntnu commented:
Do proceed – I couldn't quite think through the implications but looks like @niranjchandrasekaran is good with this, so 👍 from me too.

niranjchandrasekaran committed:
add aggregate

niranjchandrasekaran committed:
change pathlib to os.path

niranjchandrasekaran committed:
correct sql url

niranjchandrasekaran committed:
fix path of sql file
